ingest-app/src/cmr/ingest/data/granule_bulk_update.clj

```diff
@@ -0,0 +1,253 @@
+(ns cmr.ingest.data.granule-bulk-update
+  "Stores and retrieves granule bulk update status and task information."
+  (:require
+   [cheshire.core :as json]
+   [clojure.java.jdbc :as j]
+   [cmr.common.lifecycle :as lifecycle]
+   [cmr.common.log :refer (debug info warn error)]
+   [cmr.common.services.errors :as errors]
+   [cmr.common.util :refer [defn-timed] :as util]
+   [cmr.ingest.config :as config]
+   [cmr.oracle.connection]
+   [cmr.oracle.connection :as oracle]
+   [cmr.oracle.sql-utils :as su]))
+
+(defn- generate-failed-msg
+ "Generate the FAILED part of the status message."
+ [num-failed-granules]
+ (when (not= 0 num-failed-granules)
+   (str num-failed-granules " FAILED")))
+
+(defn- generate-skipped-msg
+ "Generate the SKIPPED part of the status message."
+ [num-failed-granules num-skipped-granules num-total-granules]
+ (when (not= 0 num-skipped-granules)
+   (if (not= 0 num-failed-granules)
+     (if (not= num-total-granules (+ num-failed-granules num-skipped-granules))
+       (str ", " num-skipped-granules " SKIPPED")
+       (str " and " num-skipped-granules " SKIPPED"))
+     (str num-skipped-granules " SKIPPED"))))
```

Comment:
This branching string logic may be a bit over-engineered - these functions could all be one-liners (or eliminated) if we change the final message to:

> Task completed with `(0..X)` skipped, `(0..Y)` failed, and `(0..Z)` successfully updated out of `total` total granule update(s)

Since this only applies for tasks with one or more non-successful granule updates, a full census might actually be preferable, since it would help users pin down exactly what caused the non-successes. Although, I concede that a message like `10 skipped, 0 failed` could be confusing... just food for thought.

Reply:
I'm just following the convention set by the collection bulk update. I think if we were to change that message we would need to consult our users